en-US/win10/ArduinoWiringPortingGuide.md

```diff
@@ -0,0 +1,356 @@
+
+
+
+## Common Problems
+
+<a name="prob_missingiot"></a>
+
+### Can't find "Arduino Wiring Application" Visual C++ project template in Visual Studio
+
+**Cause**: The Windows IoT Project Templates extension for Visual Studio is not installed.
+
+**Solution**: You must install the Visual Studio Extension for Windows IoT Project Templates before you can create Arduino Wiring projects in Visual Studio. Head over to [Windows IoT Core Project Templates extension page](https://visualstudiogallery.msdn.microsoft.com/55b357e1-a533-43ad-82a5-a88ac4b01dec) to download the extension from the Visual Studio Gallery!
+
+<a name="prob_hardwareserial"></a>
+
+### ERROR: 'unresolved external symbol "class HardwareSerial Serial"'
+
+**Cause**: This issue occurs when there are `Serial` references in your Arduino Wiring sketches or libraries.
+
+**Solution**: Use the "File" and "Line" fields on this error to locate the reference(s) causing the issue, and then use the <a href="#port_serial">Removing References to "Serial"</a> section of this page to resolve the issue.
+
+<a name="prob_identifier"></a>
+
+### ERROR: "identifier not found" when calling a function
+
+**Cause**: This error occurs during the linker process when a function is invoked that has not yet been declared in the document.
+
+**Solution**: In C++, all functions must be declared before they are invoked. If you have defined a new function in your sketch file, either the declaration or the entire implementation of the function must be above any attempts to invoke it (typically at the top of the document).
+
+**Example**:
+
+The following block of code will raise the error "'myFunction': identifier not found"
+
+{% highlight C++ %}
+
+void setup()
+{
+
+}
+
+void loop()
+{
+    myFunction();
+}
+
+void myFunction()
+{
+    //do something
+}
+
+{% endhighlight %}
+
+There are two solutions. First, you may declare the function above any invocations. Typically, this declaration is done at the top of the file.
+
+{% highlight C++ %}
+
+void myFunction();
+
+void setup()
+{
+
+}
+
+void loop()
+{
+    myFunction();
+}
+
+void myFunction()
+{
+    //do something
+}
+
+{% endhighlight %}
+
+The other option is to move the entire implementation of the function above any invocations. This has the effect of both declaring and defining the function at the same time.
+
+{% highlight C++ %}
+
+void setup()
+{
+
+}
+
+void myFunction()
+{
+    //do something
+}
+
+void loop()
+{
+    myFunction();
+}
+
+{% endhighlight %}
+
+<a name="prob_hanging"></a>
+
+### My solution hangs infinitely when being initialized
+
+There is a known issue which can cause a C++ solution to hang infinitely (deadlock) when being initialized. If you find that your solution appears to have this behavior and you are unable to use the debugger to 'break in' to any statement in the setup() or loop() sections of your Arduino Wiring application, you may be experiencing this type of hanging issue.
```

Comment:
if "your solution appears to have this behavior, ... you may be experiencing this" seems a bit wordy/redundant.